examples/lock_exchange.jl

```diff
@@ -0,0 +1,63 @@
+using Oceananigans
```

Comment:
Can we add a description of what this example hopes to achieve, similar to the other examples?